crates/nargo_cli/src/cli/compile_cmd.rs

```diff
@@ -125,13 +123,30 @@ pub(crate) fn compile_circuit<B: Backend>(
     compile_options: &CompileOptions,
 ) -> Result<CompiledProgram, CliError<B>> {
     let mut context = resolve_root_manifest(program_dir)?;
-    let result = compile_main(
-        &mut context,
+    let result = compile_main(&mut context, compile_options);
+    let mut program = report_errors(result, &context, compile_options.deny_warnings)?;
+
+    // Apply backend specific optimizations.
```

Comment:
not a blocker: It's perhaps not intuitive that compile applies backend specific optimizations, or the fact that compile needs a backend to begin with

Reply:
Yeah, my preference is that we don't immediately optimise everything coming out of the noir compiler immediately. For the purposes of this PR however I want to avoid making huge changes to how nargo uses a `CompiledProgram`.

In future we can maybe do more interesting things such as `nargo compile` writes a generic artifact to disk which won't get recompiled unless the code changes, etc.

Reply:
makes sense